### ISSUE TITLE
final package names

### ISSUE THREAD

**Issue by tomwayson** [Esri/cedar#279]
I think we should use scoped (@esri/*) packages.

See https://github.com/ArcGIS/rest-js/issues/9 for relevant discussion.

**Comment by benstoltz** [comment]
@tomwayson If other OS repos are moving in that direction then I definitely agree. Similar naming conventions ftw

**Comment by tomwayson** [comment]
For package names I'm thinking:
```
@esri/cedar
@esri/cedar-amcharts
```

The question is what to name the global(s) output by the UMD build?

In https://github.com/ArcGIS/arcgis-rest-js/issues/9 we learned that rollup can output all the packages to a single global namespace. Cedar currently uses the `Cedar()` constructor as a global and a namespace for a few static methods. I don't know if rollup will allow the namespace be a constructor. I'm not keen on that anyway. I'd rather use `cedar` or `cdr` as a namespace and have it expose things like `let c = new cedar.Chart(...);` and/or `let c = cedar.newChart(...);`. The thing will be that cedar-amchart's `renderSpec()` would have to be named `renderAmchartSpec()` b/c once we add other engines they'd share the same namesapce. 

The alternativ would be to have each live in it's own namespace so:

```
@esri/cedar - > cedar
@esri/cedar-amcharts - > cedarAmcharts
```

But I personally prefer a single namespace.

**Comment by tomwayson** [comment]
So turns out @benstoltz already [had this set up to use the constructor as the global namespace](https://github.com/Esri/cedar/blob/d55aebb788a7c6a851b4abb7a08637fe209feabe/packages/cedar/profiles/base.js#L19).

Also I've discovered that rollup is quite flexible in terms of how we append to that global namespace. For example, @esri/cedar-amcharts can append all of it's exports under `.amCharts`. So, I think we should plan to use just a single global namespace that we append to.

Now that we know what's possible, the last question is what name should we use for that global namespace and what should it be. As I said above, I'm not keen on using a constructor as the namespace, especially if we're going to append to it. Also, I'm not keen on using a namespace as a constructor. I think `new Cedar(...)` is confusing. What's a "Cedar"? Am I creating a tree?

I'd prefer using `cedar` as the global namespace and have it be nothing more than a namespace to hold everything else, like a more meaningful as a constructor: `var chart = new cedar.Chart(...)` or 

```
import { Chart } from 'cedar'
let chart = new Chart(...)
```
@ajturner @benstoltz does that work for you?

I know we're probably out of mental bandwidth for yet another philosophical discussion, but I think this is the last one we need to have before finalizing the v1 API, package structure/naming, and build output so we can get on w/ releasing this thing.

**Comment by tomwayson** [comment]
See how ugly it is using a constructor as a namespace?

https://github.com/Esri/cedar/pull/298/files#diff-26cc732ea6460bd0edc9badf7430a91eR11

**Comment by ajturner** [comment]
Using `Chart` sounds reasonable

On Oct 7, 2017, at 6:04 PM, Tom Wayson <notifications@github.com<mailto:notifications@github.com>> wrote:


So turns out @benstoltz<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_benstoltz&d=DwMFaQ&c=n6-cguzQvX_tUIrZOS_4Og&r=k9u7PfHvjHW9lzI6dYwFzg&m=_rSDzHKeDtzYZm6GrXFzlPsZHkVyYTiPKlYFKsFkogs&s=0t624dU_At4Iq4MghcPQeCnBWJVBygrTekVAA09PJKo&e=> already had this set up to use the constructor as the global namespace<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Esri_cedar_blob_d55aebb788a7c6a851b4abb7a08637fe209feabe_packages_cedar_profiles_base.js-23L19&d=DwMFaQ&c=n6-cguzQvX_tUIrZOS_4Og&r=k9u7PfHvjHW9lzI6dYwFzg&m=_rSDzHKeDtzYZm6GrXFzlPsZHkVyYTiPKlYFKsFkogs&s=ju7Y2yvOWUsPBDLdLANHfQtXyNKciFE6deqZjiv8mPQ&e=>.

Also I've discovered that rollup is quite flexible in terms of how we append to that global namespace. For example, @esri/cedar-amcharts can append all of it's exports under .amCharts. So, I think we should plan to use just a single global namespace that we append to.

Now that we know what's possible, the last question is what name should we use for that global namespace and what should it be. As I said above, I'm not keen on using a constructor as the namespace, especially if we're going to append to it. Also, I'm not keen on using a namespace as a constructor. I think new Cedar(...) is confusing. What's a "Cedar"? Am I creating a tree?

I'd prefer using cedar as the global namespace and have it be nothing more than a namespace to hold everything else, like a more meaningful as a constructor: var chart = new cedar.Chart(...) or

import { Chart } from 'cedar'
let chart = new Chart(...)


@ajturner<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ajturner&d=DwMFaQ&c=n6-cguzQvX_tUIrZOS_4Og&r=k9u7PfHvjHW9lzI6dYwFzg&m=_rSDzHKeDtzYZm6GrXFzlPsZHkVyYTiPKlYFKsFkogs&s=rvkXIUExw1SR2cg-OAW5ou0874DTsoix7dds0i8uEIY&e=> @benstoltz<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_benstoltz&d=DwMFaQ&c=n6-cguzQvX_tUIrZOS_4Og&r=k9u7PfHvjHW9lzI6dYwFzg&m=_rSDzHKeDtzYZm6GrXFzlPsZHkVyYTiPKlYFKsFkogs&s=0t624dU_At4Iq4MghcPQeCnBWJVBygrTekVAA09PJKo&e=> does that work for you?

I know we're probably out of mental bandwidth for yet another philosophical discussion, but I think this is the last one we need to have before finalizing the v1 API, package structure/naming, and build output so we can get on w/ releasing this thing.

**Comment by benstoltz** [comment]
@tomwayson using `Chart` sounds good to me.

**Comment by tomwayson** [comment]
Thx, I'll make a new PR after #299 is merged.

**Comment by tomwayson** [comment]
resolved in #302
